apis/v1beta1/config.go

```diff
@@ -179,6 +179,68 @@ func (c *Config) getRbacRulesForComponentKinds(logger logr.Logger, componentKind
 	return rules, nil
 }
 
+// getRbacRolesForComponentKinds gets the RBAC Roles for the given ComponentKind(s).
+func (c *Config) getRbacRolesForComponentKinds(logger logr.Logger, otelCollectorName string, componentKinds ...ComponentKind) ([]*rbacv1.Role, error) {
```

Comment:
these methods are starting to really really bloat this file. I think it's time we figure out a better way to the delegation here. I'm thinking we could make a method that wraps the iteration logic and the enabled components logic and then is passed in the parser method that should be called. LMK if you want to pair on this.

Reply:
It totally makes sense to me.